src/mprispropertiesadaptor.cpp

```diff
@@ -0,0 +1,241 @@
+ *
+ */
+
+#include "mprisplayer_p.h"
+#include "mprispropertiesadaptor_p.h"
+
+using namespace Amber;
+
+/*
+ * Implementation of adaptor class MprisPlayerAdaptor
+ */
+
+MprisPropertiesAdaptor::MprisPropertiesAdaptor(MprisPlayerPrivate *parent)
+    : QDBusAbstractAdaptor(parent)
+    , m_playerPrivate(parent)
+{
+    setAutoRelaySignals(true);
+}
+
+MprisPropertiesAdaptor::~MprisPropertiesAdaptor()
+{
+}
+
+void MprisPropertiesAdaptor::ReplyPropertyNotFoundError(const QString &interface_name, const QString &property_name)
+{
+    // The property was not found
+    m_playerPrivate->sendErrorReply(QDBusError::UnknownProperty,
+                                    QString::fromLatin1("Property %1%2%3 was not found in object %4")
+                                    .arg(interface_name,
+                                         QString::fromLatin1(interface_name.isEmpty() ? "" : "."),
+                                         property_name, m_playerPrivate->message().path()));
+}
+
+void MprisPropertiesAdaptor::ReplyPropertyReadOnlyError(const QString &interface_name, const QString &property_name)
+{
+    // The property is read only
+    m_playerPrivate->sendErrorReply(QDBusError::PropertyReadOnly,
+                                    QString::fromLatin1("Property %1.%2 is read-only")
+                                    .arg(interface_name, property_name));
+}
+
+void MprisPropertiesAdaptor::ReplyInternalError()
+{
+    // The property is read only
+    m_playerPrivate->sendErrorReply(QDBusError::InternalError,
+                                    QString::fromLatin1("Internal error"));
+}
+
+// Format: {DBus property name, MprisPlayer property name or MprisPlayerPrivate getter() method}
+static const QMap<QString, QString> playerGetMap{
+    {"CanControl", "canControl"},
+    {"CanGoNext", "canGoNext"},
+    {"CanGoPrevious", "canGoPrevious"},
+    {"CanPause", "canPause"},
+    {"CanPlay", "canPlay"},
+    {"CanSeek", "canSeek"},
+    {"LoopStatus", "loopStatus()"},
+    {"MaximumRate", "maximumRate"},
+    {"Metadata", "metaData()"},
+    {"MinimumRate", "minimumRate"},
+    {"PlaybackStatus", "playbackStatus()"},
+    {"Position", "position()"},
+    {"Rate", "rate"},
+    {"Shuffle", "shuffle"},
+    {"Volume", "volume"},
+};
+
+static const QMap<QString, QString> serviceGetMap{
+    {"CanQuit", "canQuit()"},
+    {"Fullscreen", "fullscreen()"},
+    {"CanSetFullscreen", "canSetFullscreen()"},
+    {"CanRaise", "canRaise()"},
+    {"HasTrackList", "hasTrackList()"},
+    {"Identity", "identity()"},
+    {"DesktopEntry", "desktopEntry()"},
+    {"SupportedUriSchemes", "supportedMimeTypes()"},
+    {"SupportedMimeTypes", "supportedUriSchemes()"},
+};
+
+QDBusVariant MprisPropertiesAdaptor::Get(const QString &interface_name, const QString &property_name)
+{
+    QMap<QString, QString> getMap;
+    QVariant result;
+
+    if (interface_name == QStringLiteral("org.mpris.MediaPlayer2.Player")) {
```

Comment:
I guess QLatin1String() is usually used in these cases (I guess it mostly just takes less memory)

Reply:
Yes, true. Thanks, I've changed it.